app/controllers/auth0_controller.rb

```diff
@@ -4,6 +4,13 @@ def callback
     # and the IdP
     session[:userinfo] = request.env["omniauth.auth"]
 
+    # DEBUG: Is the session set correctly on the auth0 callback? Is it ready to
+    # read straight away or is there a delay?
+    if Rails.env.production?
+      Rails.logger.info("* Auth0 callback contained the following id: #{request.env.dig("omniauth.auth", "uid")}")
```

Comment:
I'm trying to use dig here because as this code is going to develop it should be safe to run in production before I can revert it.